### PR TITLE
Avoid using -r requirements for installs

### PR DESCRIPTION
**Pull request by gadfort**
@aulihant this solves the issue of the lint on 3.6 with the new docs build. It also makes it a little easier to install since before we asked for two install steps and now can just do one

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,8 +92,8 @@ To install the project from source (recommended for developers only).
 ```bash
 git clone https://github.com/siliconcompiler/siliconcompiler
 cd siliconcompiler
-pip install -r requirements.txt
 python -m pip install -e .
+python -m pip install -e .[docs,test]  # Install dependencies for generating docs and running tests
```
Comment (nmoroze) [inline README.md:96]: Could technically skip the bare `python -m pip install -e .`, unless the idea is to give someone a path forward if the docs/test deps install fail?

**Review by nmoroze — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (gadfort) [reply README.md:96]: The idea was to give regular users the basic choice and developers the added sections. Maybe that is not 100% clear

Comment (aulihant) [reply README.md:96]: I think it _would_ be more clear to update to this:
```python -m pip install -e .              # Required install step```
```python -m pip install -e .[docs,test]  # Optional install step for generating docs and running tests```

**Review by aulihant — APPROVED** [review]
Confirmed instructions work; updates to verbiage on what is required and what is optional install step is a nice to have, but not a deal breaker for me.

**Pull request merged**